client/my-sites/plans-features-main/hooks/use-generate-action.ts

```diff
@@ -0,0 +1,191 @@
+		isLargeCurrency?: boolean;
+		isStuck?: boolean;
+		planSlug: PlanSlug;
+		planTitle?: string;
+		priceString?: string;
+		selectedStorageAddOn?: AddOns.AddOnMeta | null;
+	} ) => {
+		let actions: PlanActionOverrides = {
+			currentPlan: {},
+			loggedInFreePlan: {},
+			trialAlreadyUsed: {},
+		};
+
+		/* 1. Launch Page actions */
+		if ( isLaunchPage ) {
+			let text = translate( 'Select %(plan)s', {
+				args: {
+					plan: planTitle,
+				},
+				context: 'Button to select a paid plan by plan name, e.g., "Select Personal"',
+				comment:
+					'A button to select a new paid plan. Check screenshot - https://cloudup.com/cb_9FMG_R01',
+			} );
+
+			if ( isFreePlan( planSlug ) ) {
+				text = translate( 'Keep this plan', {
+					comment:
+						'A selection to keep the current plan. Check screenshot - https://cloudup.com/cb_9FMG_R01',
+				} );
+			} else if ( isStuck && ! isLargeCurrency ) {
+				text = translate( 'Select %(plan)s – %(priceString)s', {
+					args: {
+						plan: planTitle,
+						priceString: priceString ?? '',
+					},
+					comment:
+						'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Select Premium - $10',
+				} );
+			}
+
+			return { text };
+		}
+
+		/* 2. Onboarding actions */
+		if ( isInSignup ) {
+			let text = translate( 'Get %(plan)s', {
+				args: {
+					plan: planTitle,
+				},
+			} );
+			let postButtonText = null;
+
+			if ( isFreeTrialAction ) {
+				text = translate( 'Try for free' );
+			} else if ( isFreePlan( planSlug ) ) {
+				text = translate( 'Start with Free' );
+			} else if ( isStuck && ! isLargeCurrency ) {
+				text = translate( 'Get %(plan)s – %(priceString)s', {
+					args: {
+						plan: planTitle,
+						priceString: priceString ?? '',
+					},
+					comment:
+						'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Get Premium - $10',
+					// TODO: Revisit this type and why we have to force inference of string
+				} ) as string;
+			} else if ( isStuck && isLargeCurrency ) {
+				text = translate( 'Get %(plan)s {{span}}%(priceString)s{{/span}}', {
+					args: {
+						plan: planTitle,
+						priceString: priceString ?? '',
+					},
+					comment:
+						'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Get Premium - $10',
+					components: {
+						// span: <span className="plan-features-2023-grid__actions-signup-plan-text" />,
+					},
+				} );
+			}
+
+			if (
+				isBusinessPlan( planSlug ) &&
+				! eligibleForFreeHostingTrial &&
+				intentFromProps === 'plans-new-hosted-site'
+			) {
+				postButtonText = translate( "You've already used your free trial! Thanks!" );
+			}
+
+			return {
+				callback: getActionCallback( { planSlug, cartItemForPlan, selectedStorageAddOn } ),
+				postButtonText,
+				// TODO: Revisit status
+				status: 'enabled',
+				text,
+			};
+		}
+
+		/* 3. Current plan actions */
```

Comment:
**Note:**

The code below reflects CTA data for a spotlight plan. We will be refactoring to match the patterns in L78 and L108